__tests__/app.test.js

```diff
@@ -158,6 +159,59 @@ describe("/api", () => {
               });
           });
         });
+        describe("POST", () => {
+          test("201: sends an object of the posted comment", () => {
+            return request(app)
+              .post("/api/articles/5/comments")
+              .send({ username: "butter_bridge", body: "this is a comment!" })
+              .expect(201)
+              .then(({ body }) => {
+                const { comment } = body;
+                expect(typeof comment.comment_id).toBe("number");
+                expect(comment.author).toBe("butter_bridge");
+                expect(comment.body).toBe("this is a comment!");
+                expect(typeof comment.created_at).toBe("string");
+                expect(comment.votes).toBe(0);
+                expect(comment.article_id).toBe(5);
+              });
+          });
+          test("400: sends an appropriate error if article id is invalid", () => {
+            return request(app)
+              .post("/api/articles/hello/comments")
+              .send({ username: "butter_bridge", body: "this is a comment!" })
+              .expect(400)
+              .then(({ body }) => {
+                expect(body.msg).toBe("Bad Request");
+              });
+          });
+          test("404: sends an appropriate error if article id is valid but doesn't exist", () => {
+            return request(app)
+              .post("/api/articles/744859587/comments")
+              .send({ username: "butter_bridge", body: "this is a comment!" })
+              .expect(404)
+              .then(({ body }) => {
+                expect(body.msg).toBe("Article not found");
+              });
+          });
+          test("400: sends an appropriate error if the comment body is empty", () => {
```

Comment:
Good thinking here, although technically our data type is valid even if it's empty so would you want the server to decide this is an error? 
You might want to consider something like invalid data types or missing keys instead to be a server error.